examples/nucleo_g474re/ads101x/main.cpp

```diff
@@ -0,0 +1,184 @@
+                PT_CALL(adc.startSingleShotConversion(modm::ads101x::InputMultiplexer::Input4));
+                while (PT_CALL(adc.isBusy()))
+                {
+                    PT_YIELD();
+                }
+
+                PT_CALL(adc.readConversionResult());
+                MODM_LOG_INFO << "Single Conversion result in range 2.048V: " << data.getVoltage() << modm::endl;
+
+                // Test single shoot mode with full scale range 4.096
+                PT_CALL(adc.setFullScaleRange(modm::ads101x::FullScaleRange::V4_096));
+                PT_CALL(adc.startSingleShotConversion(modm::ads101x::InputMultiplexer::Input4));
+                while (PT_CALL(adc.isBusy()))
+                {
+                    PT_YIELD();
+                }
+
+                PT_CALL(adc.readConversionResult());
+                MODM_LOG_INFO << "Single Conversion result in range 4.096V: " << data.getVoltage() << modm::endl;
+
+                // Test single shoot mode with full scale range 6.144
+                PT_CALL(adc.setFullScaleRange(modm::ads101x::FullScaleRange::V6_144));
+                PT_CALL(adc.startSingleShotConversion(modm::ads101x::InputMultiplexer::Input4));
+                while (PT_CALL(adc.isBusy()))
+                {
+                    PT_YIELD();
+                }
+
+                PT_CALL(adc.readConversionResult());
+                MODM_LOG_INFO << "Single Conversion result in range 6.144V: " << data.getVoltage() << modm::endl;
+
+                shortTimeout.restart(std::chrono::milliseconds(1000));
+                PT_WAIT_UNTIL(shortTimeout.isExpired());
+            }
+
+            // Continuous mode test
+            MODM_LOG_INFO << "Started continuous conversion test" << modm::endl;
+            PT_CALL(adc.setFullScaleRange(modm::ads101x::FullScaleRange::V6_144));
+            PT_CALL(adc.startContinuousConversion(modm::ads101x::InputMultiplexer::Input4));
+
+            timeout.restart(std::chrono::seconds(10));
+            while (not timeout.isExpired())
+            {
+                shortTimeout.restart(std::chrono::milliseconds(1000));
+                PT_WAIT_UNTIL(shortTimeout.isExpired());
+                PT_CALL(adc.readConversionResult());
+                MODM_LOG_INFO << "Continuous Conversion result in range 6.144V: " << data.getVoltage() << modm::endl;
+            }
+        }
+        PT_END();
+    }
+
+private:
+    modm::ads101x::Data data;
+    modm::Ads101x<I2cMaster> adc;
+
+    modm::Timeout timeout;
+    modm::ShortTimeout shortTimeout;
+
+} adcThread;
+
+int
+main()
+{
+    Board::initialize();
+
+    Scl::configure(modm::platform::Gpio::InputType::PullUp);
+    Sda::configure(modm::platform::Gpio::InputType::PullUp);
+    Scl::configure(modm::platform::Gpio::OutputType::OpenDrain);
+    Sda::configure(modm::platform::Gpio::OutputType::OpenDrain);
+
+    I2cMaster::connect<Scl::Scl, Sda::Sda>(I2cMaster::PullUps::Internal);
```

Comment:
The gpio modes are also configured inside `connect()`.

```suggestion
    I2cMaster::connect<Scl::Scl, Sda::Sda>(I2cMaster::PullUps::Internal);
```